substrate/frame/revive/src/exec.rs

```diff
@@ -1668,17 +1641,12 @@ where
 	}
 
 	fn set_immutable_data(&mut self, data: ImmutableData) -> Result<(), DispatchError> {
-		if self.top_frame().entry_point == ExportedFunction::Call {
+		let frame = self.top_frame_mut();
+		if frame.entry_point == ExportedFunction::Call || data.is_empty() {
```

Comment:
Since it is now valid to call it multiple times, should empty data just clear it? Not sure if it is worth it though (since contracts can maintain the immutable data internally and call this once before returning anyways).

Reply:
So we would need to remove the storage item then? Because if we just remove the check we have the situation where we have immutable data of length zero and no immutable data as two different cases. I like to avoid that.